File: viz/src/domain/annotations/circle.js
// src/domain/annotation/circle.js
import * as d3 from 'd3';

export const Circle = class {
    constructor(x, y, r) {
        this.x = x;
        this.y = y;
        this.r = r;
        this.svg = d3.select("#canvas").select("svg");
        this.lineSource = {
            lineX: this.x + this.r,
            lineY: this.y
        };
    }

    // display () {
    //     this.svg.append("g")
    //         .append("circle")
    //         .attr("x", this.cx)
    //         .attr("y", this.cy)
    //         .attr("r", this.r)
    //         .style("fill", '#00000F')
    //         .style("opacity", 0.3);
    // }
};

export function createCircle(x, y, r) {
    return new Circle(x, y, r);
}

File: system/src/domain/viz/circle.py
from src.domain.viz.shape import Shape


class Circle(Shape):
    def __init__(self, x, y, r):
        self.type = "circle"
        self.x = x
        self.y = y
        self.r = r

    def to_dict(self):
        return {
            'type': self.type,
            'x': self.x,
            'y': self.y,
            'r': self.r
        }